### PR TITLE
Dynamics Updates

### PR DESCRIPTION
**Pull request by nandantumu**
Fixed bug in dynamics, integrated function into single track model, added type hinting to agents, and updated failing test with results directly from the commonroad implementation.

### CODE REVIEW

**Comment by luigiberducci** [comment]
@nandantumu please remove RK4 warning

File: gym/f110_gym/envs/base_classes.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 Replacement of the old RaceCar, Simulator classes in C++
 Author: Hongrui Zheng
 """
+from typing import List
```
Comment (luigiberducci) [inline gym/f110_gym/envs/base_classes.py:29]: Starting with Python 3.7, this allows to use the standard collections directly for type hinting.

```suggestion
from __future__ import annotations
```

File: gym/f110_gym/envs/base_classes.py [hunk 2]
```diff
@@ -406,7 +407,7 @@ def __init__(
         self.ego_idx = ego_idx
         self.params = params
         self.agent_poses = np.empty((self.num_agents, 3))
-        self.agents = []
+        self.agents: List[RaceCar] = []
```
Comment (luigiberducci) [inline gym/f110_gym/envs/base_classes.py:410]:
```suggestion
        self.agents: list[RaceCar] = []
```

File: gym/f110_gym/envs/dynamic_models.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 Track dynamic model
 Following the implementation of commanroad's Single Track Dynamics model
 Original implementation: https://gitlab.lrz.de/tum-cps/commonroad-vehicle-models/
-Author: Hongrui Zheng
+Author: Hongrui Zheng, Renukanandan Tumu
```
Comment (luigiberducci) [inline gym/f110_gym/envs/dynamic_models.py:27]: Nothing against. 
Shall we put credits in each file or remove them from all files and update the license?

**Review by luigiberducci — CHANGES_REQUESTED** [review]
Minor change on the type hint. Looks fine to me.

File: gym/f110_gym/envs/dynamic_models.py (hunk 3 above)
Comment (nandantumu) [reply gym/f110_gym/envs/dynamic_models.py:27]: I agree, we should either burn all of them or update all of them. Git blame exists, so I'm in support of burning all of them.

**Review by hzheng40 — APPROVED** [review]

File: gym/f110_gym/envs/dynamic_models.py (hunk 3 above)
Comment (luigiberducci) [reply gym/f110_gym/envs/dynamic_models.py:27]: Move discussion to next meeting

**Review by luigiberducci — APPROVED** [review]

**Pull request merged**